src/main/java/dev/slimevr/vr/processor/skeleton/HumanSkeleton.java

```diff
@@ -841,22 +819,34 @@ protected void updateLocalTransforms() {
 			trackerWaistNode.localTransform.setRotation(rotBuf1);
 		} else if (hmdTracker != null) {
 			// Align spine yaw with HMD
+			hmdTracker.getRotation(rotBuf1);
 			rotBuf1.fromAngles(0, rotBuf1.getYaw(), 0);
+
 			neckNode.localTransform.setRotation(rotBuf1);
-			chestNode.localTransform.setRotation(rotBuf1);
 			trackerChestNode.localTransform.setRotation(rotBuf1);
+			chestNode.localTransform.setRotation(rotBuf1);
 			waistNode.localTransform.setRotation(rotBuf1);
 			hipNode.localTransform.setRotation(rotBuf1);
 			trackerWaistNode.localTransform.setRotation(rotBuf1);
 		}
 
 		// Left Leg
-		rotBuf1.loadIdentity();
-		rotBuf2.loadIdentity();
-		if (leftUpperLegTracker != null)
+
+		// Get rotations
+		if (leftUpperLegTracker != null) {
 			leftUpperLegTracker.getRotation(rotBuf1);
-		if (leftLowerLegTracker != null)
+		} else {
+			// Align with HMD's yaw
```

Comment:
Why? It should be from previous tracker, not from HMD

Reply:
Yeah didn't think about it